src/utilcode/clrhost_nodependencies.cpp

```diff
@@ -383,6 +394,10 @@ FastFreeInProcessHeapFunc __ClrFreeInProcessHeap = (FastFreeInProcessHeapFunc) C
 
 const NoThrow nothrow = { 0 };
 
+#ifdef HAS_ADDRESS_SANITIZER
+// use standard heap functions for address santizier
```

Comment:
Now I actually wonder if we cannot keep the "new" operators unchanged. The reason is that they will end up calling malloc and so the sanitizer should be able to handle it at that level, I guess.
See this stack trace:
```
#0  __GI___libc_malloc (bytes=140737488347536) at malloc.c:2876
#1  0x00007ffff66d7548 in CorUnix::InternalMalloc (szSize=72) at /home/janvorli/git/src/src/pal/src/cruntime/malloc.cpp:114
#2  0x00007ffff66d74de in PAL_malloc (szSize=72) at /home/janvorli/git/src/src/pal/src/cruntime/malloc.cpp:98
#3  0x00007ffff675a674 in HeapAlloc (hHeap=0x1020304, dwFlags=0, numberOfBytes=72) at /home/janvorli/git/src/src/pal/src/memory/heap.cpp:242
#4  0x00007ffff5e721d2 in EEHeapAlloc (hHeap=0x1020304, dwFlags=0, dwBytes=56) at /home/janvorli/git/src/src/vm/hosting.cpp:593
#5  0x00007ffff5e72343 in EEHeapAllocInProcessHeap (dwFlags=0, dwBytes=56) at /home/janvorli/git/src/src/vm/hosting.cpp:635
#6  0x00007ffff5c019da in ClrAllocInProcessHeapBootstrap (dwFlags=0, dwBytes=56) at /home/janvorli/git/src/src/utilcode/clrhost_nodependencies.cpp:351
#7  0x00007ffff5c02c5f in ClrAllocInProcessHeap (dwFlags=0, dwBytes=...) at /home/janvorli/git/src/src/inc/clrhost.h:212
#8  0x00007ffff5c01df2 in operator new (n=56) at /home/janvorli/git/src/src/utilcode/clrhost_nodependencies.cpp:447
```